FirebaseAdmin/FirebaseAdmin/Messaging/LightSettings.cs

```diff
@@ -0,0 +1,194 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+using FirebaseAdmin.Messaging.Util;
+using Newtonsoft.Json;
+
+namespace FirebaseAdmin.Messaging
+{
+    /// <summary>
+    /// Represents light settings in an Android Notification.
+    /// </summary>
+    public sealed class LightSettings
+    {
+        private LightSettingsColor lightColor = new LightSettingsColor();
+
+        /// <summary>
+        /// Gets or sets the color value of the light settings.
+        /// </summary>
+        [JsonIgnore]
+        public string Color
+        {
+            get
+            {
+                return this.lightColor.ColorString();
+            }
+
+            set
+            {
+                this.lightColor = LightSettingsColor.FromString(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the light on duration in milliseconds.
+        /// </summary>
+        [JsonIgnore]
+        public long LightOnDurationMillis { get; set; }
+
+        /// <summary>
+        /// Gets or sets the light off duration in milliseconds.
+        /// </summary>
+        [JsonIgnore]
+        public long LightOffDurationMillis { get; set; }
+
+        /// <summary>
+        /// Gets or sets the light settings color representation as accepted by the FCM backend service.
+        /// </summary>
+        [JsonProperty("color")]
+        private LightSettingsColor LightColor
+        {
+            get
+            {
+                return this.lightColor;
+            }
+
+            set
+            {
+                this.lightColor = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string representation of <see cref="LightOnDurationMillis"/>.
+        /// </summary>
+        [JsonProperty("light_on_duration")]
+        private string LightOnDurationMillisString
+        {
+            get
+            {
+                return TimeConverter.LongMillisToString(this.LightOnDurationMillis);
+            }
+
+            set
+            {
+                this.LightOnDurationMillis = TimeConverter.StringToLongMillis(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the string representation of <see cref="LightOffDurationMillis"/>.
+        /// </summary>
+        [JsonProperty("light_off_duration")]
+        private string LightOffDurationMillisString
+        {
+            get
+            {
+                return TimeConverter.LongMillisToString(this.LightOffDurationMillis);
+            }
+
+            set
+            {
+                this.LightOffDurationMillis = TimeConverter.StringToLongMillis(value);
+            }
+        }
+
+        /// <summary>
+        /// Copies this Light Settings, and validates the content of it to ensure that it can be
```

Comment:
I think "this Light Settings" should be one of:

-- these light settings
-- this `LightSettings` object

Suggest omitting "of it" to keep it simpler.

Reply:
Thanks! Updated.